packages/plugins/geth/src/devtxs.ts

```diff
@@ -15,20 +15,23 @@ export default class DevTxs {
     this.logger = embark.logger;
     this.events = embark.events;
   }
+
   public async init() {
-    const { rpcHost, rpcPort, wsRPC, wsHost, wsPort } = this.blockchainConfig;
-    if (wsRPC) {
-      this.web3 = new Web3('ws://' + wsHost + ':' + wsPort);
-    } else {
-      this.web3 = new Web3(rpcHost + ':' + rpcPort);
+    if (!this.shouldStartDevTxs()) {
+      return;
     }
+    this.web3 = new Web3(this.blockchainConfig.endpoint);
 
     const accounts = await this.web3.eth.getAccounts();
     this.web3.eth.defaultAccount = accounts[0];
 
     this.registerConsoleCommands();
   }
 
+  private shouldStartDevTxs() {
+    return (this.blockchainConfig.enabled && this.blockchainConfig.clientConfig && this.blockchainConfig.clientConfig.miningMode === 'dev');
```

Comment:
It would be even better if we used the config object directly (ie `this.config.blockchainConfig.enabled`)
That way, if the config changes, this condition would use the newest values